etna/whatson/models.py

```diff
@@ -0,0 +1,621 @@
+        return self.name
+
+
+@register_snippet
+class AudienceType(models.Model):
+    """
+    This snippet model is used so that editors can add audience types,
+    which we use via the audience_type ForeignKey to add audience types
+    to event pages.
+    """
+
+    name = models.CharField(
+        max_length=255,
+        verbose_name=_("name"),
+    )
+
+    slug = models.SlugField(
+        max_length=255,
+        verbose_name=_("slug"),
+        unique=True,
+    )
+
+    class Meta:
+        verbose_name = _("Audience type")
+        verbose_name_plural = _("Audience types")
+
+    def __str__(self):
+        return self.name
+
+
+class AudienceTypeOrderable(Orderable):
+    """
+    This model is used to add multiple audience types to event pages.
+    """
+
+    page = ParentalKey(
+        "wagtailcore.Page",
+        on_delete=models.CASCADE,
+        related_name="event_audience_types",
+    )
+
+    audience_type = models.ForeignKey(
+        "whatson.AudienceType",
+        on_delete=models.CASCADE,
+        related_name="audience_types",
+    )
+
+
+@register_snippet
+class AccessType(models.Model):
+    """
+    This snippet model is used so that editors can add access types,
+    which we use via the AccessTypeOrderable to add multiple access
+    types to event pages.
+    """
+
+    name = models.CharField(
+        max_length=255,
+        verbose_name=_("name"),
+    )
+
+    slug = models.SlugField(
+        max_length=255,
+        verbose_name=_("slug"),
+        unique=True,
+    )
+
+    class Meta:
+        verbose_name = _("Access type")
+        verbose_name_plural = _("Access types")
+
+    panels = [
+        FieldPanel("name"),
+    ]
+
+    def __str__(self):
+        return self.name
+
+
+class AccessTypeOrderable(Orderable):
+    """
+    This model is used to add multiple access types to event pages.
+    """
+
+    page = ParentalKey(
+        "wagtailcore.Page",
+        on_delete=models.CASCADE,
+        related_name="event_access_types",
+    )
+
+    access_type = models.ForeignKey(
+        "whatson.AccessType",
+        on_delete=models.CASCADE,
+        related_name="access_types",
+    )
+
+
+class Host(Orderable):
+    """
+    This model is used to add host information to event pages.
+    """
+
+    page = ParentalKey(
+        "wagtailcore.Page",
+        on_delete=models.CASCADE,
+        related_name="hosts",
+    )
+
+    name = models.CharField(
+        max_length=100,
+        verbose_name=_("name"),
+        blank=False,
+    )
+
+    description = models.CharField(
+        max_length=200,
+        verbose_name=_("description"),
+        blank=True,
+    )
+
+    image = models.ForeignKey(
+        get_image_model_string(),
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name="+",
+    )
+
+    panels = [
+        FieldPanel("name"),
+        FieldPanel("description"),
+        FieldPanel("image"),
+    ]
+
+
+class Speaker(Orderable):
```

Comment:
`EventSpeaker` might make for a more sensible name here, as they are tied to event pages